src/LinuxTracing/LeafFunctionCallManager.cpp

```diff
@@ -130,6 +130,11 @@ Callstack::CallstackType LeafFunctionCallManager::PatchCallerOfLeafFunction(
   // If the caller is not executable, we have an unwinding error.
   unwindstack::MapInfo* caller_map_info = current_maps->Find(libunwindstack_leaf_caller_pc);
   if (caller_map_info == nullptr || (caller_map_info->flags() & PROT_EXEC) == 0) {
+    // If the error was because the stack sample was too small, the user can act and increase the
+    // stack size. So we report that case separately.
+    if (stack_size > stack_dump_size_) {
```

Comment:
Soo, I need to understand before approving: the stack dump is too small, but libunwindstack tries to unwind anyway, reports the new frame, but makes a mistake that we detect looking at the `MapInfo`? Is that the case?

Reply:
Yes, exactly this is happening.

Reply:
The point is, that assuming $rbp is untouched ("stack_size" would be two complete frames), we might be still completely fine when unwinding the first frame even if "stack_size" is larger than the "stack dump size".

Reply:
Ok, so this is a case of speculative frame that might or might not be correct?

Reply:
Yes, I think so.